yarn-project/ethereum/src/utils.ts

```diff
@@ -80,7 +92,43 @@ export function prettyLogViemErrorMsg(err: any) {
   return err?.message ?? err;
 }
 
-export function formatViemError(error: any): string {
+/**
+ * Formats a Viem error into a FormattedViemError instance.
+ * @param error - The error to format.
+ * @param abi - The ABI to use for decoding.
+ * @returns A FormattedViemError instance.
+ */
+export function formatViemError(error: any, abi: Abi = RollupAbi): FormattedViemError {
+  // First try to decode as a custom error using the ABI
+  try {
+    if (error?.data) {
+      // Try to decode the error data using the ABI
+      const decoded = decodeErrorResult({
+        abi,
+        data: error.data as Hex,
+      });
+      if (decoded) {
+        return new FormattedViemError(`${decoded.errorName}(${decoded.args?.join(', ') ?? ''})`, error?.metaMessages);
+      }
+    }
+
+    // If it's a BaseError, try to get the custom error through ContractFunctionRevertedError
+    if (error instanceof BaseError) {
+      const revertError = error.walk(err => err instanceof ContractFunctionRevertedError);
+      if (revertError instanceof ContractFunctionRevertedError) {
+        const errorName = revertError.data?.errorName ?? '';
```

Comment:
Can we extract the error selector here if we don't find the `errorName`?

Reply:
huh, I did have a case for that where we extract `.signature`, probably lost in the sea of commits 🤦 
Good catch, will add back